vms/platformvm/state/chain_time_helpers.go

```diff
@@ -64,42 +75,93 @@ func GetNextStakerChangeTime(state Chain, defaultTime time.Time) (time.Time, err
 		}
 
 		time := it.Value().NextTime
-		if time.Before(defaultTime) {
-			defaultTime = time
+		if time.Before(nextTime) {
+			nextTime = time
 		}
 	}
-	return defaultTime, nil
+
+	return getNextSoVEvictionTime(config, state, nextTime)
+}
+
+func getNextSoVEvictionTime(
+	config validatorfee.Config,
+	state Chain,
+	nextTime time.Time,
+) (time.Time, error) {
+	sovIterator, err := state.GetActiveSubnetOnlyValidatorsIterator()
+	if err != nil {
+		return time.Time{}, fmt.Errorf("could not iterate over active SoVs: %w", err)
+	}
+	defer sovIterator.Release()
+
+	// If there are no SoVs, return
+	if !sovIterator.Next() {
+		return nextTime, nil
+	}
+
+	// Calculate the remaining funds that the next validator to evict has.
+	var (
+		// GetActiveSubnetOnlyValidatorsIterator iterates in order of increasing
+		// EndAccumulatedFee, so the first SoV is the next SoV to evict.
+		sov         = sovIterator.Value()
+		accruedFees = state.GetAccruedFees()
+	)
+	remainingFunds, err := math.Sub(sov.EndAccumulatedFee, accruedFees)
+	if err != nil {
+		return time.Time{}, fmt.Errorf("could not calculate remaining funds: %w", err)
+	}
+
+	// Calculate how many seconds the remaining funds can last for.
+	var (
+		currentTime = state.GetTimestamp()
+		maxSeconds  = uint64(nextTime.Sub(currentTime) / time.Second)
+	)
+	feeState := validatorfee.State{
+		Current: gas.Gas(state.NumActiveSubnetOnlyValidators()),
+		Excess:  state.GetSoVExcess(),
+	}
+	remainingSeconds := feeState.SecondsRemaining(
```

Comment:
Just want to make sure we handle the case where a validator has remainingFunds for a fractional second properly.
It seems SecondsRemaining rounds down where the actual eviction time would be the next second, but I am not sure.

Reply:
Added tests for fractional seconds.